### PR TITLE
Add Delete Branch

### PR DESCRIPTION
**Pull request by DavidTanner**
Add the request to delete a branch.  This is useful after merging a pull request.

My use case is in the Github Pull Request Builder plugin for Jenkins I would like to delete the branch after merging the pull request.

### CODE REVIEW

File: src/main/java/org/kohsuke/github/GHBranch.java [hunk 1]
```diff
@@ -42,6 +44,21 @@ public String getSHA1() {
     public String toString() {
         return "Branch:" + name + " in " + owner.getUrl();
     }
+    
+    protected String getApiRoute() {
+        return "/repos/"+owner.getOwnerName()+"/"+owner.getName()+"/git/refs/heads/" + name;
```
Comment (KostyaSha) [inline src/main/java/org/kohsuke/github/GHBranch.java]: use one coding style, with spaces everywhere or without

File: src/main/java/org/kohsuke/github/GHBranch.java [hunk 2]
```diff
@@ -42,6 +44,21 @@ public String getSHA1() {
...
+    }
+    
+    /**
+     * Delete a branch
+     * 
+     * Tells github to delete the branch from the repository.
+     *
```
Comment (KostyaSha) [inline src/main/java/org/kohsuke/github/GHBranch.java]: free line

File: src/main/java/org/kohsuke/github/GHBranch.java [hunk 3]
```diff
@@ -42,6 +44,21 @@ public String getSHA1() {
...
     }
+    
+    protected String getApiRoute() {
+        return "/repos/"+owner.getOwnerName()+"/"+owner.getName()+"/git/refs/heads/" + name;
+    }
+    
+    /**
+     * Delete a branch
+     * 
+     * Tells github to delete the branch from the repository.
```
Comment (KostyaSha) [inline src/main/java/org/kohsuke/github/GHBranch.java]: tautology

**Comment by KostyaSha** [comment]
:-1: PR useless according to github-api documentation this functionality exist in https://developer.github.com/v3/git/refs/ and implemented in `GHRef.class`. You can get `GHRef` from `GHRepository.class` and then call `GHRef.delete()`

File: src/main/java/org/kohsuke/github/GHBranch.java (hunk 1 above)
Comment (DavidTanner) [inline src/main/java/org/kohsuke/github/GHBranch.java]: I can do that.  This line was copy pasted.

**Comment by DavidTanner** [comment]
Well isn't that just the bees knees.  Thank you.

**Pull request closed without merging**